src/participant.rs

```diff
@@ -110,6 +155,56 @@ pub(crate) trait ProtocolParticipant {
         main_storage: &mut Storage,
     ) -> Result<ProcessOutcome<Self::Output>>;
 
+    /// Returns a list of all participant IDs, including `self`'s.
+    fn all_participants(&self) -> Vec<ParticipantIdentifier> {
+        std::iter::once(&self.id())
```

Comment:
How do you feel about this code that doesn't use iterators:
```rust
    let mut participant = self.other_ids().clone();
        participant.push(self.id());
        participant
```

Reply:
I feel positive! I was prematurely optimizing to avoid reallocating in case the cloned capacity is exactly the length and we have to reallocate, but in retrospect, it's not worth the readability hit.